### ISSUE TITLE
Type Overrides with TypeScript

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/pg-promise#756]
The latest version of the driver had some extensions to let you set isolated type parsers on the connection pool, i.e. effectively on the `db` object level, rather than setting them globally.

Although it works fine as is in JavaScript, TypeScript declarations need to be updated.

**Comment by vitaly-t** [comment]
This `type override` feature in the driver is a bit of a hack, and requires direct JavaScript code import, using `require`.

Below is a complete example of how such custom type parsers can be set on the pool level.

```ts
import * as pgLib from 'pg-promise';
import {ITypeOverrides} from 'pg-promise/typescript/pg-subset';

const TypeOverrides = require('pg-promise/node_modules/pg/lib/type-overrides');

const types: ITypeOverrides = new TypeOverrides();

types.setTypeParser(20, BigInt);

const pgp = pgLib({/* initialization options */});

const db = pgp({
    host: 'localhost',
    port: 5432,
    // ...other connection details

    types // our pool-level type parsers
});
```

**Comment by vitaly-t** [comment]
This has been released in version 10.7.0 of the library.

**Comment by vitaly-t** [comment]
In order to get rid of that awkward `require` approach above, I have created the following PR:

[Exposing TypeOverrides from the library's root](https://github.com/brianc/node-postgres/pull/2369).

Once it is merged and released, the above code example will be improved.
